mne/utils/config.py

```diff
@@ -486,7 +487,16 @@ def sys_info(fid=None, show_paths=False):
         PyQt5:         5.15.0
     """  # noqa: E501
     ljust = 15
-    out = 'Platform:'.ljust(ljust) + platform.platform() + '\n'
+    platform_str = platform.platform()
+    if platform.system() == 'Darwin' and sys.version_info[:2] < (3, 8):
+        # platform.system() in Python < 3.8 doesn't call
+        # platform.mac_ver() if we're on Darwin, so we don't get a nice macOS
+        # version number. Therefore, let's do this manually here.
+        macos_ver = platform.mac_ver()[0]
+        macos_architecture = re.match('Darwin-.*?-(.*)', platform_str)[1]
```

Comment:
I am hesitant to add `re.match` without any check that the result is not None

Reply:
@larsoner Better @ 7436826 ?